software/28.Gin/README.md

```diff
@@ -0,0 +1,154 @@
+```shell
+go mod init workshop-gin
+```
+
+Afterwards you can follow the official [documentation](https://github.com/gin-gonic/gin#installation) to install Gin correctly.
+>Once again don't hesitate if you have any issue during the installation ! :smile:
+
+
+**Now everything should work properly you can start the workshop !**
+
+## Step 1: The codebase
+
+We designed a boilerplate for this project, it will be useful to start it, but also to summarize all the required parts of an API
+We will have :
+
+- `controllers`: this is where you will design your routes endpoints. it will often be a wrapper that calls other API or the database itself
+- `middlewares`: those are a type of function used to intercept and process informations between two functions
+- `routes`: the core of the router, setting up all the routes handler, with their middlewares and controllers
+
+There are plenty of other important package you may need in a real API, but these are the main one you can be sure you'll need one time or another. Now let's code.
+
+Add a route on the endpoint `/`:
+- It uses the `GET` method
+- It must call a helloWorld function that responds `"hello: world"` in a json format
+
+> You can test it with `http://localhost:8080/` in a browser or with `postman`
+
+## Step 2: Let's get started
+
+You are now fine with the creation of endpoints.
+Next step is to create different request method.
+There is many [request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), you already experimented the `GET` request.
+
+For this step, we will use some others' method:
+- a `GET` request `/item/template`: to get the template of the item as follows.
+```json
+{
+  "name": "name",
+  "description": "description",
+  "price": 0
+}
+```
+
+- a `POST` request `/item/display`: to display the item passed in the body as follows.
+  <img src="./.assets/request-item-display.png">
+
+
+also try to implement a default endpoint for the URL's that doesn't match any routes. (404 notfound error)
+
+
+
+> This [tutorial](https://blog.logrocket.com/making-http-requests-in-go/) may help you
+
+> You should create a structure item with the following fields:
+> - name
+> - description
+> - price
+>
+> and bind the request body to the structure to retrieve it (do it in a structure folder).
+<details>
+  <summary>See how to request with postman :satellite:</summary>
+
+Enter your URL and the method you which to use in the title bar and click `Send`.
+
+![Seek](../../.github/go-http/seek.png)
+
+Then the result (if there is any) will be printed out at the bottom.
+
+![Result](../../.github/go-http/result.png)
+
+</details>
+
+## Step 3: Authorization middleware
+
+Now let's create a middleware `Auth()` that will check if the user is authorized to access the endpoint.
+
+We won't use real token like a [JWT](https://jwt.io/), we will just check if the request contains a header with the name `Authorization` and a value.
```

Comment:
You need to explain what is a middleware and why use it. Also you need to add documentation about how to use it.